numbat/src/ast.rs

```diff
@@ -85,6 +85,13 @@ pub enum Expression {
     Boolean(Span, bool),
     String(Span, Vec<StringPart>),
     Condition(Span, Box<Expression>, Box<Expression>, Box<Expression>),
+    MakeStruct {
```

Comment:
This is not super important, but I think it could be useful to change the naming a bit to make it easier to understand the code. I would distinguish between a *struct*, which is the type-level construct, and an *instance* of the struct, which is the actual object constructed at runtime. 

I think I would rename:
 * `MakeStruct` => `InstantiateStruct`
 * `AccessStruct` => `AccessField`